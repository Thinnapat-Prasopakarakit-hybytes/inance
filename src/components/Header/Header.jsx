import React, { useState } from "react";
import { FaPhoneAlt, FaEnvelope } from "react-icons/fa";
import { NavLink } from "react-router-dom";
import "./Header.scss";

const Header = () => {
  const [isOpen, setIsOpen] = useState(false);

  const toggleNavbar = () => {
    setIsOpen(!isOpen);
  };

  return (
    <header className="header_section">
      <div className="header_top">
        <div className="container-fluid">
          <div className="contact_nav">
            <NavLink to="/">
              <FaPhoneAlt className="contact_nav-icon" />
              <span className="contact_nav-text">Call : +01 123455678990</span>
            </NavLink>
            <NavLink to="/">
              <FaEnvelope className="contact_nav-icon" />
              <span className="contact_nav-text">Email : demo@gmail.com</span>
            </NavLink>
          </div>
        </div>
      </div>
      <div className="header_bottom">
        <div className="container-fluid">
          <nav className="navbar navbar-expand-lg custom_nav-container">
            <NavLink to="/" className="navbar-brand">
              <span>Inance</span>
            </NavLink>

            {
              // Bootstrap
              /*
              <button
                className="navbar-toggler"
                type="button"
                data-bs-toggle="collapse"
                data-bs-target="#navbarNav"
                aria-controls="navbarNav"
                aria-expanded="false"
                aria-label="Toggle navigation"
              >
                <span className="navbar-toggler-icon"></span>
              </button>

              <div className="collapse navbar-collapse" id="navbarNav"> */
            }

            <button
              className="navbar-toggler"
              type="button"
              aria-controls="navbarNav"
              aria-expanded={isOpen}
              aria-label="Toggle navigation"
              onClick={toggleNavbar}
            >
              <span className="navbar-toggler-icon"></span>
            </button>

            <div className={`collapse navbar-collapse ${isOpen ? "show" : ""}`}>
              <ul className="navbar-nav">
                <li className="nav-item">
                  <NavLink to="/" className="nav-link" activeClassName="active">
                    Home <span className="visually-hidden">(current)</span>
                  </NavLink>
                </li>
                <li className="nav-item">
                  <NavLink
                    to="/about"
                    className="nav-link"
                    activeClassName="active"
                  >
                    About
                  </NavLink>
                </li>
                <li className="nav-item">
                  <NavLink
                    to="/service"
                    className="nav-link"
                    activeClassName="active"
                  >
                    Services
                  </NavLink>
                </li>
                <li className="nav-item">
                  <NavLink
                    to="/contact"
                    className="nav-link"
                    activeClassName="active"
                  >
                    Contact Us
                  </NavLink>
                </li>
              </ul>
            </div>
          </nav>
        </div>
      </div>
    </header>
  );
};

export default Header;
