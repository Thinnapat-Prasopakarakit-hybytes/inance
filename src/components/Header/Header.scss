@use "../../styles/partials/variables" as *;
@use "../../styles/partials/mixins" as *;

.hero_area {
  position: relative;
  background-color: $background;
}

.header_section {
  .container-fluid {
    padding: 0 25px;
  }

  .header_top {
    padding: 15px 0;
    background-color: $black;

    .contact_nav {
      display: flex;
      justify-content: space-between;

      a {
        color: $white;
        text-decoration: none;
        display: flex;
        align-items: center;
        gap: 5px;
      }
      &-icon {
        color: $primary2;
        font-size: 1rem;
      }
      &-text {
        color: $white;
        display: inline-block;
        vertical-align: middle;
      }
    }
  }

  .header_bottom {
    padding: 10px 0;

    .navbar-toggler {
      outline: none;
      padding: 0;
      width: 37px;
      height: 42px;
      transition: transform 0.35s ease;
      border: none;

      &:focus {
        box-shadow: none;
      }
      span {
        display: block;
        width: 35px;
        height: 4px;
        background-color: $black;
        margin: 7px 0;
        border-radius: 5px;
        transition: transform 0.35s ease;
        position: relative;

        &::before,
        &::after {
          content: "";
          position: absolute;
          left: 0;
          height: 100%;
          width: 100%;
          background-color: $black;
          border-radius: 5px;
          transition: transform 0.35s ease;
        }

        &::before {
          top: -10px;
        }

        &::after {
          top: 10px;
        }
      }

      &[aria-expanded="true"] {
        transform: rotate(360deg);

        span {
          transform: rotate(45deg);

          &::before,
          &::after {
            transform: rotate(90deg);
            top: 0;
          }
        }
      }

      &[aria-expanded="false"] {
        span {
          &.s-1,
          &.s-2,
          &.s-3 {
            transform: none;
          }
        }
      }
    }
  }
}

.navbar-collapse {
  max-height: 0;
  overflow: visible;
  display: flex;
  display: block !important;
  &.show {
    max-height: 500px;
  }

  @include desktop {
    max-height: 0;
    overflow: hidden;
    transition: max-height 0.35s ease-in-out;
  }
}

.navbar-brand {
  padding: 0;
  margin: 0;
  color: $black;
  font-weight: bold;
  font-size: 24px !important;

  span {
    color: $primary1;
    text-transform: uppercase;
  }
}

.custom_nav-container {
  padding: 0 !important;
  display: flex;
  align-items: center;
  margin-left: auto;

  .navbar-nav {
    margin-left: auto;

    .nav-item {
      .nav-link {
        padding: 10px 25px;
        color: $black;
        text-align: center;

        &:hover,
        &.active {
          color: $primary1;
        }
      }
    }
  }
}

.language {
  margin: 0 10px;
  display: inline-flex;
  align-items: center;
  gap: 10px;
  color: $white;
  .active {
    color: $primary1;
    color: #9e9d9d;

    @include tablet {
      display: none;
    }
  }

  .separator {
    @include tablet {
      display: none;
    }
  }

  .toggle {
    color: $primary2;
    cursor: pointer;

    &:hover {
      text-decoration: underline;
    }
  }
}
