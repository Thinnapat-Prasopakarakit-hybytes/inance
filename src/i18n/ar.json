{
  "header": {
    "phone": "اتصل : +01 123455678990",
    "email": "البريد الإلكتروني : demo@gmail.com"
  },
  "nav": {
    "home": "الرئيسية",
    "about": "من نحن",
    "services": "خدماتنا",
    "contact": "اتصل بنا"
  },
  "feature": {
    "repair": "إصلاح",
    "improve": "تحسين",
    "maintain": "صيانة"
  },
  "slider": {
    "firstLine": "إصلاح و",
    "secondLine": "صيانة",
    "thirdLine": "خدمات",
    "description": "لوريم إيبسوم دولور سيت أميت، كونسيكتيتور أديباسينغ إليت. نيمو إنيم، فيريتات أوديو، ليبريتاس لوريم إيبسوم إنتيجريت. مينيما كوي أوت أتشينت إينيم أوديو أكوربوز، سوسيب تينيت أورناريه إلتراتو نيمو ريسيتات. إني أمبيد كوي أتالوغ إيرابيل، إيس أونجتوت أوديو لوريم. أوديا فان إينيس ديوس غريلانديسكو فوريس، أرتيموس كانايور.",
    "contact": "اتصل بنا"
  },
  "about": {
    "title": "من نحن",
    "readMore": "اقرأ المزيد",
    "description": "هناك حقيقة مثبتة منذ زمن طويل وهي أن المحتوى المقروء لصفحة ما سيلهي القارئ عن التركيز على الشكل الخارجي للنص."
  },
  "professional": {
    "title": "نحن نقدم خدمات",
    "subtitle": "منزلية احترافية",
    "description": "هناك حقيقة مثبتة منذ زمن طويل وهي أن المحتوى المقروء لصفحة ما سيلهي القارئ عن التركيز على الشكل الخارجي للنص.",
    "readMore": "اقرأ المزيد"
  },
  "services": {
    "title": "خدماتنا",
    "maintenance": "صيانة",
    "electrical": "كهرباء",
    "plumbing": "سباكة",
    "description": "هناك حقيقة مثبتة منذ زمن طويل وهي أن المحتوى المقروء لصفحة ما سيلهي القارئ",
    "viewMore": "عرض المزيد"
  },
  "contact": {
    "title": "اتصل بنا",
    "name": "الاسم",
    "phone": "رقم الهاتف",
    "email": "البريد الإلكتروني",
    "message": "الرسالة",
    "send": "إرسال"
  },
  "client": {
    "title": "ماذا يقول عملاؤنا",
    "name": "جورش موريك",
    "testimonial": "يقوم بتقسيم النص إلى أجزاء حسب الضرورة، مما يجعله أول مولد حقيقي على الإنترنت. يستخدم قاموسًا يحتوي على أكثر من 200 كلمة لاتينية، مع عدد قليل من تراكيب الجمل النموذجية، لتوليد نص لوريم إيبسوم."
  },
  "info": {
    "getInTouch": "تواصل معنا",
    "text": "لوريم إيبسوم هو نص تجريدي بسيط",
    "followUs": "تابعنا"
  },
  "footer": {
    "copyright": "جميع الحقوق محفوظة لـ",
    "company": "قوالب HTML المجانية"
  },
  "notFound": {
    "text": "الصفحة غير موجودة",
    "description": "عذراً! الصفحة التي تبحث عنها غير موجودة",
    "link": "العودة إلى الصفحة الرئيسية"
  }
}
