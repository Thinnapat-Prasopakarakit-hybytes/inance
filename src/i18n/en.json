{
  "header": {
    "phone": "Call : +01 123455678990",
    "email": "Email : demo@gmail.com"
  },
  "nav": {
    "home": "Home",
    "about": "About",
    "services": "Services",
    "contact": "Contact Us"
  },
  "feature": {
    "repair": "Repair",
    "improve": "Improve",
    "maintain": "Maintain"
  },
  "slider": {
    "firstLine": "Repair and",
    "secondLine": "Maintenance",
    "thirdLine": "Services",
    "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Qui harum voluptatem adipisci. Quos molestiae saepe dicta nobis pariatur, tempora iusto, ad possimus soluta hic praesentium mollitia consequatur beatae, aspernatur culpa.",
    "contact": "Contact Us"
  },
  "about": {
    "title": "About us",
    "readMore": "Read More",
    "description": "There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable."
  },
  "professional": {
    "title": "We Provide Professional",
    "subtitle": "Home Services.",
    "description": "randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All randomised words which don't look even slightly",
    "readMore": "Read More"
  },
  "services": {
    "title": "Our Services",
    "maintenance": "Maintenance",
    "electrical": "Electrical",
    "plumbing": "Plumbing",
    "description": "when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal",
    "viewMore": "View More"
  },
  "contact": {
    "title": "Contact Us",
    "name": "Name",
    "phone": "Phone Number",
    "email": "Email",
    "message": "Message",
    "send": "SEND"
  },
  "client": {
    "title": "What Our Clients Say",
    "name": "Jorch morik",
    "testimonial": "chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum"
  },
  "info": {
    "getInTouch": "Get In Touch",
    "text": "Lorem Ipsum is simply dummy text",
    "followUs": "Follow Us"
  },
  "footer": {
    "copyright": "All Rights Reserved By",
    "company": "Free Html Templates"
  },
  "notFound": {
    "text": "Page not found",
    "description": "Oops! The page you are looking for does not exist",
    "link": "Back to Home"
  }
}
