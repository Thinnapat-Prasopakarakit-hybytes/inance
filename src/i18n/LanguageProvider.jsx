import { useState, createContext, useEffect } from "react";
import { IntlProvider } from "react-intl";
import { useLocation, useParams } from "react-router-dom";
import en from "./en.json";
import ar from "./ar.json";
import { HelmetProvider } from "react-helmet-async";
import SEO from "../components/SEO";
import { logEvent, logLanguageChange } from "../analytics";

export const LanguageContext = createContext();

const messages = {
  en,
  ar,
};

const LanguageProvider = ({ children }) => {
  const { lang } = useParams();
  const { pathname } = useLocation();
  const [locale, setLocale] = useState(lang === "ar" ? "ar" : "en");

  useEffect(() => {
    if (lang && (lang === "en" || lang === "ar")) {
      localStorage.setItem("lang", lang);
      setLocale(lang);
      document.documentElement.dir = lang === "ar" ? "rtl" : "ltr";
      document.documentElement.lang = lang;
    }
  }, [lang]);

  const handleLanguageChange = (newLocale) => {
    if (locale !== newLocale) {
      logEvent("Language", "Change", newLocale);
      const pathWithoutLang = pathname.replace(/^\/(en|ar)/, "");
      const newPath =
        pathWithoutLang === "" || pathWithoutLang === "/"
          ? `/${newLocale}`
          : `/${newLocale}${pathWithoutLang}`;

      window.location.href = newPath;
    }
  };

  return (
    <HelmetProvider>
      <LanguageContext.Provider value={{ locale, handleLanguageChange }}>
        <IntlProvider
          messages={messages[locale]}
          locale={locale}
          defaultLocale="en"
        >
          <SEO lang={locale} />
          {children}
        </IntlProvider>
      </LanguageContext.Provider>
    </HelmetProvider>
  );
};

export default LanguageProvider;
