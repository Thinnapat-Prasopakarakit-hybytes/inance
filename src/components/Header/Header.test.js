import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Header from "./Header";
import { LanguageContext } from "../../i18n/LanguageProvider";
import { IntlProvider } from "react-intl";
import en from "../../i18n/en.json";
import ar from "../../i18n/ar.json";

const messages = {
  en,
  ar,
};

const renderHeaderComponent = (locale = "en") => {
  return render(
    <BrowserRouter>
      <LanguageContext.Provider value={{ locale }}>
        <IntlProvider messages={messages[locale]} locale={locale}>
          <Header />
        </IntlProvider>
      </LanguageContext.Provider>
    </BrowserRouter>
  );
};

describe("Header Component", () => {
  it("should renders brand logo", () => {
    renderHeaderComponent();
    expect(screen.getByText("Inance")).toBeInTheDocument();
  });

  it("should renders contact information", () => {
    renderHeaderComponent();
    expect(screen.getByText("Call : +01 123455678990")).toBeInTheDocument();
    expect(screen.getByText("Email : demo@gmail.com")).toBeInTheDocument();
    expect(screen.getByTestId("phone-icon")).toBeInTheDocument();
    expect(screen.getByTestId("envelope-icon")).toBeInTheDocument();
  });

  it("should render navigation links", () => {
    renderHeaderComponent();
    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(screen.getByText("About")).toBeInTheDocument();
    expect(screen.getByText("Services")).toBeInTheDocument();
    expect(screen.getByText("Contact Us")).toBeInTheDocument();
  });

  it("should renders toggle button for mobile navigation", () => {
    renderHeaderComponent();
    const toggleButton = screen.getByRole("button", {
      name: /toggle navigation/i,
    });
    expect(toggleButton).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = renderHeaderComponent();
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Navigation Toggle Functionality", () => {
  it("should toggles the navbar on button click", () => {
    renderHeaderComponent();
    const toggleButton = screen.getByLabelText("Toggle navigation");
    const navbarCollapse = screen.getByTestId("navbar-collapse");

    expect(navbarCollapse).not.toHaveClass("show");
    fireEvent.click(toggleButton);
    expect(navbarCollapse).toHaveClass("show");
    fireEvent.click(toggleButton);
    expect(navbarCollapse).not.toHaveClass("show");
  });

  it("should update aria-expanded attribute on toggle button", () => {
    renderHeaderComponent();
    const toggleButton = screen.getByRole("button", {
      name: /toggle navigation/i,
    });

    expect(toggleButton).toHaveAttribute("aria-expanded", "false");

    fireEvent.click(toggleButton);
    expect(toggleButton).toHaveAttribute("aria-expanded", "true");

    fireEvent.click(toggleButton);
    expect(toggleButton).toHaveAttribute("aria-expanded", "false");
  });
});

describe("Header Component - Arabic", () => {
  test("renders Arabic navigation items when locale is ar", () => {
    renderHeaderComponent("ar");

    expect(screen.getByText("الرئيسية")).toBeInTheDocument();
    expect(screen.getByText("من نحن")).toBeInTheDocument();
    expect(screen.getByText("خدماتنا")).toBeInTheDocument();
    expect(screen.getByText("اتصل بنا")).toBeInTheDocument();
  });

  test("language switcher shows correct active state for Arabic", () => {
    renderHeaderComponent("ar");

    const arabicToggle = screen.getByText("عربي");
    const englishToggle = screen.getByText("English");

    expect(arabicToggle).toHaveClass("active");
    expect(englishToggle).toHaveClass("toggle");
  });

  test("mobile menu toggle works with Arabic content", () => {
    renderHeaderComponent("ar");

    const toggleButton = screen.getByRole("button", {
      "aria-label": "Toggle navigation",
    });
    const navbarCollapse = screen.getByTestId("navbar-collapse");

    expect(navbarCollapse).not.toHaveClass("show");

    fireEvent.click(toggleButton);
    expect(navbarCollapse).toHaveClass("show");

    fireEvent.click(toggleButton);
    expect(navbarCollapse).not.toHaveClass("show");
  });
});
