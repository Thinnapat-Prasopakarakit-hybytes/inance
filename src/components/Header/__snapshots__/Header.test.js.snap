// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="header_section"
  >
    <div
      class="header_top"
    >
      <div
        class="container-fluid"
      >
        <div
          class="contact_nav"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <div
              data-testid="phone-icon"
            />
            <span
              class="contact_nav-text"
            >
              Call : +01 123455678990
            </span>
          </a>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <div
              data-testid="email-icon"
            />
            <span
              class="contact_nav-text"
            >
              Email : demo@gmail.com
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="header_bottom"
    >
      <div
        class="container-fluid"
      >
        <nav
          class="navbar navbar-expand-lg custom_nav-container"
        >
          <a
            aria-current="page"
            class="navbar-brand active"
            href="/"
          >
            <span>
              Inance
            </span>
          </a>
          <button
            aria-controls="navbarNav"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse "
            data-testid="navbar-collapse"
          >
            <ul
              class="navbar-nav"
            >
              <li
                class="nav-item"
              >
                <a
                  activeclassname="active"
                  aria-current="page"
                  class="nav-link active"
                  href="/"
                >
                  Home 
                  <span
                    class="visually-hidden"
                  >
                    (current)
                  </span>
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  activeclassname="active"
                  class="nav-link"
                  href="/about"
                >
                  About
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  activeclassname="active"
                  class="nav-link"
                  href="/service"
                >
                  Services
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  activeclassname="active"
                  class="nav-link"
                  href="/contact"
                >
                  Contact Us
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
